### PR TITLE
gh-78143: IDLE - replace  config dlg scale with ttk spinbox

### PR DESCRIPTION
**Pull request by roseman**
- replace scale with ttk.Spinbox
- if ttk.Spinbox not available (Tk < 8.5.9) use readonly ttk.Combobox (as both fully constrain available choices)
- note that test originally tried to set a value above the maximum range and relied on the scale adjusting it to the maximum; without adding extra validation code, neither the spinbox or the combobox will do this; given that neither allows arbitrary input, I adjusted the test to provide a value within range

* Issue: gh-78143

### CODE REVIEW

**Comment by terryjreedy** [comment]
See issue for initial review of widget in operation.

**Comment by terryjreedy** [comment]
Test failure is patchcheck trailing whitespace failure.

**Comment by E-Paine** [comment]
TBH, I wouldn't bother checking for 8.5.9 as of all the distros/BSDs listed on [pkgs.org](https://pkgs.org/download/tk), only CentOS 6 (which reaches EOL very soon) has <8.5.9 (8.5.7).

**Comment by terryjreedy** [comment]
Centos 6 and 7 seem to be the only ones without 8.6.  NetBSD offers 8.5.19 as an option.

**Comment by terryjreedy** [comment]
Centos 6 is no longer on the list at https://pkgs.org/download/tk.  Centos 7 has 8.5.13.  Centos 8 is 8.6.8.  The Pipelines failure report is not available.  I did an update merge, needed anyway, to force retesting.

File: Lib/idlelib/configdialog.py [hunk 1]
```diff
@@ -683,6 +689,7 @@ def var_changed_space_num(self, *params):
         "Store change to indentation size."
         value = self.space_num.get()
         changes.add_option('main', 'Indent', 'num-spaces', value)
+        self.after_idle(self.indent_chooser.selection_clear)
```
Comment (terryjreedy) [inline Lib/idlelib/configdialog.py:692]: Is this about the box being readonly?  It does not clear the currently selected value.

File: Lib/idlelib/configdialog.py [hunk 2]
```diff
@@ -570,9 +571,14 @@ def create_page_font_tab(self):
         indent_title = Label(
                 frame_indent, justify=LEFT,
                 text='Python Standard: 4 Spaces!')
-        self.indent_scale = Scale(
-                frame_indent, variable=self.space_num,
-                orient='horizontal', tickinterval=2, from_=2, to=16)
+        try:
+            self.indent_chooser = Spinbox(
+                    frame_indent, textvariable=self.space_num,
+                    state="readonly", from_=2, to=16)
```
Comment (terryjreedy) [inline Lib/idlelib/configdialog.py]: I want to allow 1.  I think max 10 is enough.  Will ask on idle-dev.  If 16 were actually needed, I would want to allow key entry in entry box.  Will want key entry if use for font size.
Needs width=2, at least when moved.
EDIT: Width and key entry done.

File: Lib/idlelib/configdialog.py [hunk 3]
```diff
@@ -570,9 +571,14 @@ def create_page_font_tab(self):
...
+        except TclError:
+            self.indent_chooser = Combobox(
+                    frame_indent, textvariable=self.space_num,
+                    state="readonly", values=list(range(2,17)))
```
Comment (terryjreedy) [inline Lib/idlelib/configdialog.py]: I would like to delete this, at least on 3.10+, by requiring 8.5.9+  (even that is hardly usable on Apple).  Will probably ask on pydev.

**Comment by terryjreedy** [comment]
I put the spinbox on the new Windows tab and made it writeable by using a StringVar.  I intend to merge it tomorrow.

**Pull request merged**